control-plane/agents/core/src/core/specs.rs

```diff
@@ -166,11 +187,20 @@ pub trait SpecOperations: Clone + Debug + Sized + StorableObject {
             let _ = spec.busy()?;
             if spec.state().deleted() {
                 return Ok(());
-            } else if !del_owned && spec.owned() {
-                return Err(SvcError::InUse {
-                    kind: spec.kind(),
-                    id: spec.uuid(),
-                });
+            } else if !del_owned {
+                spec.disowned_by(del_by);
```

Comment:
Nit: I'm not a big fan of the "_by" suffix. I wonder if we can find an alternative.

Reply:
Fixed as discussed